source/common/conn_pool/conn_pool_base.h

```diff
@@ -325,6 +331,9 @@ class ConnPoolImplBase : protected Logger::Loggable<Logger::Id::pool> {
   // The number of streams currently attached to clients.
   uint32_t num_active_streams_{0};
 
+  // Whether the connection pool is currently in a draining state
```

Comment:
For future readers can you add comments on what draining actually means (vs. idle)? Potentially here and also in the main interface file.